src/main/java/org/opentripplanner/gtfs/mapping/BikeAccessMapper.java

```diff
@@ -0,0 +1,76 @@
+package org.opentripplanner.gtfs.mapping;
+
+import org.onebusaway.gtfs.model.Route;
+import org.onebusaway.gtfs.model.Trip;
+import org.opentripplanner.model.BikeAccess;
+
+import java.util.function.IntSupplier;
+
+/**
+ * Model bike access for GTFS trips.
+ *
+ * The GTFS bike extensions is originally discussed at:
+ * https://groups.google.com/d/msg/gtfs-changes/QqaGOuNmG7o/xyqORy-T4y0J
+ *
+ * It proposes "route_bikes_allowed" in routes.txt and "trip_bikes_allowed" in trips.txt with the
+ * following semantics:
+ *
+ * 2: bikes allowed<br>
+ * 1: no bikes allowed<br>
+ * 0: no information (same as field omitted)<br>
+ *
+ * The values in trips.txt override the values in routes.txt.
+ *
+ * An alternative proposal is discussed in:
+ * https://groups.google.com/d/msg/gtfs-changes/rEiSeKNc4cs/gTTnQ_yXtPgJ
+ *
+ * Here, the field "bikes_allowed" is used in both routes.txt and trip.txt with the following
+ * semantics:
+ *
+ * 2: no bikes allowed<br>
+ * 1: bikes allowed<br>
+ * 0: no information (same as field omitted)<br>
+ *
+ * Here, the 0,1,2 semantics have been changed to match the convention used in the
+ * "wheelchair_accessible" field in trips.txt.
+ *
+ * A number of feeds are still using the original proposal and a number of feeds have been updated
+ * to use the new proposal. For now, we support both, using "bikes_allowed" if specified and then
+ * "trip_bikes_allowed".
+ */
+class BikeAccessMapper {
+
+    public static BikeAccess mapForTrip(Trip rhs) {
+        //noinspection deprecation
+        return map(rhs::getBikesAllowed, rhs::getTripBikesAllowed);
+    }
+
+    public static BikeAccess mapForRoute(Route rhs) {
+        //noinspection deprecation
+        return map(rhs::getBikesAllowed, rhs::getRouteBikesAllowed);
+    }
+
+    private static BikeAccess map(IntSupplier bikesAllowedSupplier, IntSupplier legacyBikesAllowedSupplier) {
+        int bikesAllowed = bikesAllowedSupplier.getAsInt();
+        int legacyBikesAllowed = legacyBikesAllowedSupplier.getAsInt();;
```

Comment:
This is a bi overkill, could be replaced with: 
```
public static BikeAccess mapForRoute(Route rhs) {
    int bikesAllowed = rhs.getBikesAllowed();
    int legacyBikesAllowed = rhs.getRouteBikesAllowed();
    
    if (bikesAllowed != 0) { ....
```
I like the variable declaration, because it make it clear to me what the two alternatives are.

Reply:
Agreed, fixed :-)

I kept the the common `mapValues()` to avoid duplicating its contents in `mapForRoute()` and `mapForTrip()`.